src/modules/tasks.js

```diff
@@ -14,32 +36,42 @@ function addTaskToDOM(task) {
   }
   taskElement.dataset.id = task.id;
 
-  const titleElement = document.createElement('span');
-  titleElement.className = 'title';
-  titleElement.contentEditable = true;
-  titleElement.textContent = task.title;
-  titleElement.addEventListener('input', () => {
-    task.title = titleElement.textContent;
-    saveTasksToLocalStorage();
-  });
+  const titleElement = createTitleElement(task.title);
+  titleElement.addEventListener('input', handleTitleInput);
 
-  const deleteButton = document.createElement('span');
-  deleteButton.className = 'del';
-  deleteButton.appendChild(document.createTextNode('Delete'));
+  const deleteButton = createDeleteButton();
   taskElement.appendChild(titleElement);
   taskElement.appendChild(deleteButton);
   taskList.appendChild(taskElement);
 }
 
 function loadTasks() {
   const data = localStorage.getItem('tasks');
-  if (data) {
-    const tasksFromStorage = JSON.parse(data);
-    tasksFromStorage.forEach(addTaskToDOM);
-    tasks = tasksFromStorage;
-  } else {
-    tasks = [];
+  tasks = data ? JSON.parse(data) : [];
+  tasks.forEach(addTaskToDOM);
+}
+
+function deleteTask(taskId) {
+  tasks = tasks.filter((task) => task.id.toString() !== taskId.toString());
+  saveTasksToLocalStorage();
+}
```

Comment:
- [ ] Good job setting up todos with random ids. However, for this project, the requirements indicate that the indexes for the new tasks are the length of the task array. 
You're to also update the indexes of each task whenever a task is removed to keep them unique. As per [this](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#:~:text=Deleting%20a%20task%20should%20update%20all%20remaining%20items%27%20indexes%2C%20so%20they%20represent%20the%20current%20list%20order%20and%20are%20unique(i.e.%20if%20you%27re%20deleting%20the%20first%20task%20index%201%20from%20the%20list%2C%20the%20index%20of%20the%20next%20task(2)%20should%20set%20to%201)..) requirement.